sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -612,6 +617,13 @@ case class DivideYMInterval(
   override def inputTypes: Seq[AbstractDataType] = Seq(YearMonthIntervalType, NumericType)
   override def dataType: DataType = YearMonthIntervalType()
 
+  @transient
+  private lazy val divideByZeroCheck: Any => Unit = right.dataType match {
```

Comment:
can we move this to `IntervalDivide` as well?

Reply:
I agree to Wenchen, let's move it to the common trait.